### ISSUE TITLE
[Bug]: Data Loss When Second Device Syncs

### ISSUE THREAD

**Issue by Rage-Quiff** [remotely-save/remotely-save#432]
### What happened?

I have two instances of data loss since upgrading this plugin yesterday. I had no such issues prior to the upgrade. 

I use Remotely Sync to sync between my Windows PC and Android smartphone via S3.

Today, I created a file on my smartphone. Call it Wednesday.md. I made several edits to it, and periodically did manual syncs, since the plugin on my smartphone is configured to sync only every 30 minutes (this is a bit of a surprise, as I thought auto sync was disabled on my smartphone and I don't recall changing it). 

Several minutes after the last sync from the smartphone, I booted my PC and opened Obisidan on it. This is set to sync on launch, and every 5 minutes. I think this is when the data loss happened. The PC synced an older copy of the new file, possibly from the very first sync of the file. I made no edits to the file on the PC. But a few minutes later, I noticed the discrepancy, and checked the copy on my smartphone to see the same thing. So I lost more than half the edits I made to the file, which were made after the initial sync. Copies of the file on both devices show a modified time stamp from about the time I launched Obsidian on the PC. Unfortunately, the file was too new to have been backed up, so I lost all of those edits.

Yesterday's data loss is a little different, and I'm not sure exactly what happened. A file created on two days ago, call it Monday.md, had multiple edits to it made throughout the day. I am fairly confident I didn't edit Monday.md on the smartphone, but I know I did manual syncs per usual on it. Then yesterday, I updated plugins on both devices, including Remotely Save. Sometime shortly thereafter, I noticed on the smartphone that the Monday.md file was there, but missing 100% of the edits I had made after its creation on the PC. On neither device was there a recent copy of the file in the Recycle Bin. I had a nightly PC backup of Monday.md that I was able to restore. A minute later, the file on the PC was reverted to the empty version right before my eyes. I figured it was overwritten by a sync from the smartphone, so I deleted it on both devices, and restored it again. 

I've now re-disabled auto sync on my smartphone and increased the frequency of my backups. 

I understand the sync works by file timestamp, but I'm still not sure why it would sync (and overwrite) older versions of the file. Versioning is not enabled on my S3 bucket.

I do have a second PC, running Linux, on which I occasionally open and sync this Obisidian vault. I haven't touched that PC in several days, so I don't recall if the vault is open on it right now.  I'm not sure if it's even set to auto sync. It would still be running the older version of the plugin. But the files that have experienced the data loss were never edited on that PC.

### What OS are you using?

Windows, Linux, Android

### What remote cloud services are you using?

S3

### Version of the plugin

0.3.34

### Version of Obsidian

1.5.3

### Using password or not

- [ ] Yes.

### Ensure no sensitive information

- [X] I ensure that no sensitive information is submitted in the issue.

**Comment by fyears** [comment]
thanks for your detailed explanation, and sorry to hear the data loss.

i suspect that there are some other plugins modifying or at least touching the files thus changing the timestamps. do you have any posdible plugins installed? moreover you could export the sync plans in settings page and to see the timestamp changes of that file across multiple syncs, to ensure that the modified timestamps are really changed.

moreover, in a few days i would release a new version introducing hashing and conflicts resolution, at least users don't brother trivial data loses.

**Comment by Rage-Quiff** [comment]
I do of course have other plugins enabled. But other than Remotely-Save, they don't do touch files in the background; just add text & formatting for convenience. The only possible exception would be the File Recovery core plugin. I forgot I had this enabled. Using this, I was able to recovery my lost data. Looking at the snapshots on both the Android smartphone and the Windows PC, I can see all the data I added to the note on the smartphone at about 8:30 AM that day, before I booted the Windows PC. Then at about 9:00 AM, the PC had a snapshot of the new file, but it was missing most of the data within the file. I don't recall if this matched an older snapshot on the smartphone, but it looked like an earlier version of the file. Then at like 9:15 AM, the smartphone had a snapshot that matched what was on the PC. So it seems at some point, data got lost after it was synced to the PC, or an older version of the file synced instead of the most current, and then that incomplete/older version was synced back to the smartphone.

In any case, I haven't had the problem since those two instances. I am watching it more carefully now, and have increased the frequency and depth of the File Recovery plugin snapshots. I should be able to provide better details if it occurs again, but hopefully this information was useful.

I haven't installed the update yet, but I will shortly. Thank you.
